src/crypto/tls/handshake_client_tls13.go

```diff
@@ -16,6 +16,44 @@ import (
 	"time"
 )
 
+// EXP_EventTLS13ClientHandshakeTimingInfo carries intra-stack durations for
+// state machine changes. It can be used for tracking metrics during a
+// connection.
+//
+// NOTE: This API is EXPERIMENTAL and subject to change.
+type EXP_EventTLS13ClientHandshakeTimingInfo struct {
+	timer                   func() time.Time
+	start                   time.Time
+	WriteClientHello        time.Duration
+	ReadServerHello         time.Duration
+	ReadEncryptedExtensions time.Duration
+	ReadCertificate         time.Duration
+	ReadCertificateVerify   time.Duration
+	ReadServerFinished      time.Duration
+	WriteClientFinished     time.Duration
+}
+
+// Name is required by the EXP_Event interface.
+func (e EXP_EventTLS13ClientHandshakeTimingInfo) Name() string {
+	return "TLS13ClientHandshakeTimingInfo"
+}
+
+func (e EXP_EventTLS13ClientHandshakeTimingInfo) elapsedTime() time.Duration {
```

Comment:
Just a suggestion: Since this struct only ever exists in the context of a `clientHandshakeStateTLS13`, I would just make `elapsedTime()` a method of `clientHandshakeStateTLS13` so that you can use `hs.c.config.Time()` to get the current time. (Similarly for the server timing info event.)

Reply:
I left this as-is for now, since it seems cleaner for the timing logic to hang off the thing storing the timing, rather than bolted onto something else.